src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/OsxIPGlobalProperties.cs

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace System.Net.NetworkInformation
+{
+    internal class OsxIPGlobalProperties : IPGlobalProperties
+    {
+        public override string DhcpScopeName
+        {
+            get
+            {
+                throw new PlatformNotSupportedException();
+            }
+        }
+
+        public override string DomainName
+        {
+            get
+            {
+                return HostInformation.DomainName;
+            }
+        }
+
+        public override string HostName
+        {
+            get
+            {
+                return HostInformation.HostName;
+            }
+        }
+
+        public override bool IsWinsProxy
+        {
+            get
+            {
+                return false;
+            }
+        }
+
+        public override NetBiosNodeType NodeType
+        {
+            get
+            {
+                return NetBiosNodeType.Unknown;
+            }
+        }
+
+        public unsafe override TcpConnectionInformation[] GetActiveTcpConnections()
+        {
+            int realCount = Interop.Sys.GetEstimatedTcpConnectionCount();
+            int infoCount = (int)(realCount * 1.5f);
```

Comment:
If this is required we should probably avoid floating point operations that on certain IoT devices will be emulated.